gems/pending/appliance_console/database_replication.rb

```diff
@@ -0,0 +1,116 @@
+                  :primary_host, :standby_host
+
+    def ask_for_cluster_name
+      self.cluster_name = just_ask("cluster name", cluster_name)
+    end
+
+    def ask_for_node_number
+      self.node_number = ask_for_integer("node number")
+    end
+
+    def ask_for_primary_database_credentials
+      self.primary_host              = ask_for_ip_or_hostname("primary database hostname or IP address", primary_host)
+      self.primary_database          = just_ask("name of the database on the primary host", primary_database)
+      self.primary_database_username = just_ask("username on the primary host", primary_database_username)
+      count = 0
+      loop do
+        count += 1
+        password1 = ask_for_password_or_none("database password on primary host", primary_database_password)
+        # if they took the default, just bail
+        break if password1 == primary_database_password
+        password2 = ask_for_password("database password on primary host again")
+        if password1 == password2
+          self.primary_database_password = password1
+          break
+        elsif count > 1 # only reprompt password once
+          raise ArgumentError, "passwords did not match"
+        else
+          say("\nThe passwords did not match, please try again")
+        end
+      end
+    end
+
+    def confirm(including_standby_host = nil)
+      clear_screen
+      say(<<-EOL)
+Replication Server Configuration
+
+        Cluster Name:              #{cluster_name}
+        Node Number:               #{node_number}
+        Primary Database:          #{primary_database}
+        Primary Database Username: #{primary_database_username}
+        Primary Database Password: #{primary_database_password}
+        Primary Host:              #{primary_host}
+        EOL
+      say("        Standby Host:              #{standby_host}") unless including_standby_host.nil?
+
+      agree("Apply this Replication Server Configuration? (Y/N): ")
+    end
+
+    def already_configured?
+      if File.exist?(REPMGR_CONFIG)
+        say("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        Logging.logger.info("Warning: File #{REPMGR_CONFIG} exists. Replication is already configured")
+        return agree("Continue with configuration? (Y/N): ")
+      end
+      true
+    end
+
+    def create_config_file(host)
+      File.open(DatabaseReplication::REPMGR_CONFIG, "w") do |f|
+        f.puts "cluster=#{cluster_name}"
+        f.puts "node=#{node_number}"
+        f.puts "node_name=#{host}"
+        f.puts "conninfo='host=#{host} user=#{primary_database_username} dbname=#{primary_database}'"
+        f.puts "use_replication_slots=1"
+        f.puts "pg_basebackup_options='--xlog-method=stream'"
+      end
+      true
+    end
+
+    # This method runs the provided block in a forked process.
+    # The provided block should invoke "exit 0" to report success and "exit non_zero" to report failure.
+    def fork_as_postgres
```

Comment:
Can we make this and maybe some other methods that should not be called externally private?